GNOME Adwaita dark window decorations

wigust opened the pull request:
GNOME Adwaita dark window decorations

![emacs-dark](https://cloud.githubusercontent.com/assets/7709598/20640129/91207a3e-b3ef-11e6-83e9-a98b15c6a674.png)
![emacs-light](https://cloud.githubusercontent.com/assets/7709598/20640130/9123bb68-b3ef-11e6-89b6-c519373abefc.png)

bmag commented on layers/+themes/dark-window/README.org:
```diff
@@ -0,0 +1,26 @@
+on the content and it's easier on the eyes.
+
+Emacs with a dark theme and the large white-ish window decoration isn't really a
+fit on my desktop, but that can be easily fixed using a bit of elisp with xprop!
```
This is a direct quote from the article, so it should be formatted in a quote block. Don't plagiarize.

bmag commented on layers/+themes/dark-window/README.org:
```diff
@@ -0,0 +1,26 @@
+Emacs with a dark theme and the large white-ish window decoration isn't really a
+fit on my desktop, but that can be easily fixed using a bit of elisp with xprop!
+
+See also:
```
This is not a "see also", it's the source of a quote.

bmag commented on layers/+themes/dark-window/packages.el:
```diff
@@ -0,0 +1,28 @@
+(defun dark-window/init-frame-cmds ()
+  (use-package frame-cmds
+    :config
+    (progn
```
This whole `progn` is a direct copy of http://nicolas.petton.fr/blog/emacs-dark-window-decoration.html, so you should a comment attributing credit to the original author of the code.

bmag commented on layers/+themes/dark-window/packages.el:
```diff
@@ -0,0 +1,28 @@
+          (call-process-shell-command
+           (concat "xprop -f _GTK_THEME_VARIANT 8u -set _GTK_THEME_VARIANT \"dark\" -name \"" frame-name "\""))))
+
+      (if (window-system)
```
Use `when` when you don't have an `else` clause.

wigust commented:
@bmag thanks. I added author and a quotation block. Also added a reason of creation in the description. It's cannot be fixed easily in GNOME like in other GNOME apps.

bmag commented:
I clicked the approve button because you fixed most of the issues, but you still need to add a comment before that `progn`.

wigust commented:
@bmag OK, added.

bmag commented on layers/+themes/dark-window/packages.el:
```diff
@@ -0,0 +1,33 @@
+    (progn
+      (defun set-selected-frame-color ()
+        (when (and (eq window-system 'x) (eq spacemacs--cur-theme 'spacemacs-dark))
+          (interactive)
```
`(interactive)` must be the first sexp in the function's body (after the docstring).

bmag commented on layers/+themes/dark-window/packages.el:
```diff
@@ -0,0 +1,33 @@
+    :config
+    ;; Based on Nicolas Petton code http://nicolas.petton.fr/blog/emacs-dark-window-decoration.html
+    (progn
+      (defun set-selected-frame-color ()
```
Please a documentation string for the function.

bmag replied:
Need to find a better name. The function doesn't set an arbitrary color, it specifically sets the decoration to dark. In fact, we'd probably want an argument to decide if the decoration should set to dark or light.

bmag commented on layers/+themes/dark-window/packages.el:
```diff
@@ -0,0 +1,33 @@
+             (concat "xprop -f _GTK_THEME_VARIANT 8u -set _GTK_THEME_VARIANT \"dark\" -name \"" frame-name "\"")))))
+
+      (when (and (eq window-system 'x) (eq spacemacs--cur-theme 'spacemacs-light))
+        (interactive)
```
`(interactive)` doesn't do anything here and should be removed.

bmag commented on layers/+themes/dark-window/packages.el:
```diff
@@ -0,0 +1,33 @@
+            (call-process-shell-command
+             (concat "xprop -f _GTK_THEME_VARIANT 8u -set _GTK_THEME_VARIANT \"dark\" -name \"" frame-name "\"")))))
+
+      (when (and (eq window-system 'x) (eq spacemacs--cur-theme 'spacemacs-light))
```
Why only if the theme is `spacemacs-light`? There might be a programmatic way to find out if the current theme has a dark or a light background.

bmag commented on layers/+themes/dark-window/packages.el:
```diff
@@ -0,0 +1,33 @@
+        (interactive)
+        (let ((frame-name (get-frame-name (selected-frame))))
+          (call-process-shell-command
+           (concat "xprop -f _GTK_THEME_VARIANT 8u -set _GTK_THEME_VARIANT \"light\" -name \"" frame-name "\"")))))))
```
Why not call `set-selected-frame-color` with an argument that will make the decoration light?

bmag commented on layers/+themes/dark-window/packages.el:
```diff
@@ -0,0 +1,33 @@
+    ;; Based on Nicolas Petton code http://nicolas.petton.fr/blog/emacs-dark-window-decoration.html
+    (progn
+      (defun set-selected-frame-color ()
+        (when (and (eq window-system 'x) (eq spacemacs--cur-theme 'spacemacs-dark))
```
The theme shouldn't matter here.

wigust commented:
Cannot find any variable or function which will tell me if theme or background is dark/light
For some reason color of borders doesn't change for **first** make-frame.
